tests/Unit/DataStructures/Test_TaggedContainers.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/TaggedContainers.hpp"
+#include "DataStructures/Tags.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Helpers/DataStructures/DataBox/TestHelpers.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace {
+namespace Tags {
+struct MyDouble : db::SimpleTag {
+  using type = double;
+};
+struct MyScalar : db::SimpleTag {
+  using type = Scalar<DataVector>;
+};
+struct MyDouble2 : db::SimpleTag {
+  using type = double;
+};
+}  // namespace Tags
+
+// Test that desired tags are retrieved.
+void test_tagged_containers() {
+  auto box = db::create<db::AddSimpleTags<Tags::MyDouble>>(1.2);
+  tuples::TaggedTuple<Tags::MyDouble2> tuple(4.4);
+  Variables<tmpl::list<Tags::MyScalar>> vars(10, 3.2);
+
+  auto retrieved_value1 = get<Tags::MyDouble>(box);
+  auto retrieved_value2 = get<Tags::MyDouble2>(tuple);
+  auto retrieved_value3 = get<Tags::MyScalar>(vars);
+  CHECK(retrieved_value1 == 1.2);
+  CHECK(retrieved_value2 == 4.4);
+  CHECK(retrieved_value3.get() == 3.2);
+}
```

Comment:
Can you also add:
```cpp
const auto box2 = db::create<db::AddSimpleTags<Tags::MyScalar>>(1.2);
CHECK(get<Tags::MyScalar>(vars, box2).get() == 3.2);
CHECK(get<Tags::MyScalar>(box2, vars).get() == 1.2);
CHECK(get<Tags::MyScalar>(tuple, box2, vars).get() == 1.2);
CHECK(get<Tags::MyScalar>(box2, tuple, vars).get() == 1.2);
CHECK(get<Tags::MyScalar>(box2, vars, tuple).get() == 1.2);

CHECK(*get<Tags::MyDouble>(make_not_null(&box) == 1.2);
CHECK(*get<Tags::MyDouble2>(make_not_null(&tuple) == 4.4);
CHECK(get<Tags::MyScalar>(make_not_null(&box)->get() == 3.2);

CHECK(get<Tags::MyScalar>(make_not_null(&vars), make_not_null(&box2)).get() == 3.2);
CHECK(get<Tags::MyScalar>(make_not_null(&box2), make_not_null(&vars)).get() == 1.2);
// Same thing as the 3-arg variant but with `not_null` here
```

Reply:
Can you also add some tests where you have the same tag in multiple containers with different values so you are sure it always gets it from the first container?

Reply:
Ah ok yeah I missed that